src/main/haskell/kore/app/exec/Main.hs

```diff
@@ -384,7 +395,7 @@ mainWithOptions
             clockSomethingIO "Executing"
             $ withLogger koreLogOptions (\logger ->
                 SMT.runSMT smtConfig
-                $ evalSimplifier logger
+                $ evalSimplifier logger repl
```

Comment:
I don't like that `Simplifier` now requires a `repl` or `noRepl` argument everywhere. I'm worried this is going to promote tighter coupling--not looser coupling. I think we should discuss this design more before we commit to it.